### PR TITLE
[Pipeline/Tooling] Report arm64 package and binary size metrics

### PR DESCRIPTION
**Pull request by Pythyu**
### What does this PR do?

Allows the datadog-agent pipeline to report the size of arm64 Linux packages & binaries. The tag `arch:arm64` or `arch:amd64` is used to differentiate reports.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .gitlab/pkg_metrics.yml [hunk 1]
```diff
@@ -51,17 +53,26 @@ send_pkg_size-a6:
   script:
     - source /root/.bashrc && conda activate ddpy3
     - mkdir -p /tmp/deb/agent
```
Comment (bkabrda) [inline .gitlab/pkg_metrics.yml:55]: Let's make the directory naming consistent - how about we call them all "{arch}-agent", e.g. `arm64-agent`, `amd64-agent` etc?

File: .gitlab/pkg_metrics.yml [hunk 2]
```diff
@@ -51,17 +53,26 @@ send_pkg_size-a6:
...
+    - mkdir -p /tmp/deb/arm-agent
     - mkdir -p /tmp/rpm/agent
+    - mkdir -p /tmp/rpm/arm-agent
     - mkdir -p /tmp/suse/agent
 
     # we silence dpkg and cpio output so we don't exceed gitlab log limit
 
     # debian
     - dpkg -x $OMNIBUS_PACKAGE_DIR/datadog-agent_6*_amd64.deb /tmp/deb/agent > /dev/null
     - DEB_AGENT_SIZE=$(du -sB1 /tmp/deb/agent | sed 's/\([0-9]\+\).\+/\1/')
```
Comment (bkabrda) [inline .gitlab/pkg_metrics.yml:65]: Similar to the directories, let's make the variable naming consistent. Let's use e.g. `AMD64_DEB_AGENT_SIZE` etc.

File: .gitlab/pkg_metrics.yml [hunk 3]
```diff
@@ -105,28 +123,31 @@ send_pkg_size-a7:
 
     - source /root/.bashrc && conda activate ddpy3
     - mkdir -p /tmp/deb/agent /tmp/deb/dogstatsd /tmp/deb/iot-agent /tmp/deb/heroku-agent
+    - mkdir -p /tmp/arm-deb/agent /tmp/arm-deb/dogstatsd /tmp/arm-deb/iot-agent
```
Comment (bkabrda) [inline .gitlab/pkg_metrics.yml:126]: Similar to what I proposed above, let's make the directory names consistent.

File: .gitlab/pkg_metrics.yml [hunk 4]
```diff
@@ -105,28 +123,31 @@ send_pkg_size-a7:
...
-            add_metric datadog.agent.package.size $(du -sB1 ${base}/agent | sed 's/\([0-9]\+\).\+/\1/') os:${os} package:agent agent:7 bucket_branch:$BUCKET_BRANCH
-            add_metric datadog.agent.package.size $(du -sB1 ${base}/dogstatsd | sed 's/\([0-9]\+\).\+/\1/') os:${os} package:dogstatsd agent:7 bucket_branch:$BUCKET_BRANCH
-            add_metric datadog.agent.package.size $(du -sB1 ${base}/iot-agent | sed 's/\([0-9]\+\).\+/\1/') os:${os} package:iot-agent agent:7 bucket_branch:$BUCKET_BRANCH
-            if [[ "$os" == "debian" ]]; then add_metric datadog.agent.package.size $(du -sB1 ${base}/heroku-agent | sed 's/\([0-9]\+\).\+/\1/') os:${os} package:heroku-agent agent:7 bucket_branch:$BUCKET_BRANCH; fi
+            add_metric datadog.agent.package.size $(du -sB1 ${base}/agent | sed 's/\([0-9]\+\).\+/\1/') os:${os} package:agent agent:7 bucket_branch:$BUCKET_BRANCH arch:${arch}
+            if [[ "$arch" == "amd64" || "$os" == "debian" ]]; then add_metric datadog.agent.package.size $(du -sB1 ${base}/dogstatsd | sed 's/\([0-9]\+\).\+/\1/') os:${os} package:dogstatsd agent:7 bucket_branch:$BUCKET_BRANCH arch:${arch}; fi
```
Comment (bkabrda) [inline .gitlab/pkg_metrics.yml:139]: AFAICS, we have dogstatsd builds for deb/arm64, deb/amd64, rpm/amd64 and suse/amd64. Shouldn't we report them all here?

File: .gitlab/pkg_metrics.yml [hunk 5]
```diff
@@ -105,28 +123,31 @@ send_pkg_size-a7:
...
+            add_metric datadog.agent.binary.size $(du -sB1 ${base}/agent/opt/datadog-agent/embedded/bin/security-agent | sed 's/\([0-9]\+\).\+/\1/') bin:security-agent os:${os} package:agent agent:7 bucket_branch:$BUCKET_BRANCH arch:${arch}
+            add_metric datadog.agent.binary.size $(du -sB1 ${base}/agent/opt/datadog-agent/embedded/bin/process-agent | sed 's/\([0-9]\+\).\+/\1/') bin:process-agent os:${os} package:agent agent:7 bucket_branch:$BUCKET_BRANCH arch:${arch}
+            add_metric datadog.agent.binary.size $(du -sB1 ${base}/agent/opt/datadog-agent/embedded/bin/system-probe | sed 's/\([0-9]\+\).\+/\1/') bin:system-probe os:${os} package:agent agent:7 bucket_branch:$BUCKET_BRANCH arch:${arch}
+            if [[ "$arch" == "amd64" || "$os" == "debian" ]]; then add_metric datadog.agent.binary.size $(du -sB1 ${base}/dogstatsd/opt/datadog-dogstatsd/bin/dogstatsd | sed 's/\([0-9]\+\).\+/\1/') bin:dogstatsd os:${os} package:dogstatsd agent:7 bucket_branch:$BUCKET_BRANCH arch:${arch}; fi
```
Comment (bkabrda) [inline .gitlab/pkg_metrics.yml:149]: Same as above - we have more dogstatsd builds than just deb/amd64

File: .gitlab/pkg_metrics.yml [hunk 6]
```diff
@@ -205,6 +235,50 @@ send_pkg_size-a7:
     # Make the job fail if at least one package is above threshold
     - if [ "${failures}" -ne "0" ]; then false; fi
 
+.check_arm64_pkg_size:
```
Comment (bkabrda) [inline .gitlab/pkg_metrics.yml:238]: This looks like a lot of duplicated code from `.check_pkg_size`. I think it should be possible to modify `.check_pkg_size` to work with an `ARCH` variable and then specify that variable on the `check_pkg_size*` jobs.

For the existing `check_pkg_size*` jobs, let's rename them to reflect that their for amd64.

**Review by bkabrda — CHANGES_REQUESTED** [review]
Thanks for the PR! It mostly looks good, but I left couple comments inline to get addressed - this is mostly about naming consistency as well as avoiding code duplication.

File: .gitlab/pkg_metrics.yml (hunk 4 above)
Comment (Pythyu) [reply .gitlab/pkg_metrics.yml:139]: Might be a copy paste error on this one, checking rn.

Comment (Pythyu) [reply .gitlab/pkg_metrics.yml:139]: Actually no, if arch is amd64 so deb/amd64, rpm/amd64 and suse/amd64 or os is debian so deb/arm64.
It should cover all the possible dogstatsd builds ?

Comment (bkabrda) [reply .gitlab/pkg_metrics.yml:139]: You're right, it actually does. Consider this conversation solved then :)

File: .gitlab/pkg_metrics.yml (hunk 5 above)
Comment (Pythyu) [reply .gitlab/pkg_metrics.yml:149]: Same as before it's an "or" not "and" so it should cover them all

File: .gitlab/pkg_metrics.yml (hunk 6 above)
Comment (Pythyu) [reply .gitlab/pkg_metrics.yml]: Should be good now I'll wait after the pipeline finishes to start an all_build pipeline, this way I'm sure nothing is broken

**Review by bkabrda — APPROVED** [review]

**Pull request merged**